### PR TITLE
Implemeted file type #116

### PR DESCRIPTION
**Pull request by emir-sehovic1**
- Implemented file type that can be created from Url, PrettyName and SizeInBytes
- Supported file formats can be configured in options

### CODE REVIEW

File: src/Nox.Types/Types/File/File.cs [hunk 1]
```diff
@@ -1,9 +1,129 @@
...
+
+    /// <summary>
+    /// Validates the <see cref="File"/> object.
+    /// </summary>
+    /// <returns>A validation result indicating whether the <see cref="File"/> object is valid or not.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        System.Uri.TryCreate(Value.Url, UriKind.Absolute, out var x);
```
Comment (dejobratic) [inline src/Nox.Types/Types/File/File.cs:107]: What is this statement used for?

**Review by dejobratic — COMMENTED** [review]
Looks good. Just a small comment.

File: src/Nox.Types/Types/File/File.cs (hunk 1 above)
Comment (emir-sehovic1) [reply src/Nox.Types/Types/File/File.cs]: That was left in by mistake. Removed it now, thanks

**Review by dejobratic — APPROVED** [review]
Good job.

File: src/Nox.Types.Abstractions/Enums/File/FileFormatType.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+/// <summary>
+/// Enumeration for file format types.
+/// </summary>
+public enum FileFormatType
+{
+    /// <summary>
+    /// Represents all file format types.
+    /// </summary>
+    All,
```
Comment (rochar) [inline src/Nox.Types.Abstractions/Enums/File/FileFormatType.cs:15]: to have this the enum must be decorated with Flags, and the All must be All = pdf | ....

**Review by rochar — CHANGES_REQUESTED** [review]
Looks good just a small comment

**Review by rochar — APPROVED** [review]

**Pull request merged**